src/State/Processor/RespondProcessor.php

```diff
@@ -150,4 +165,21 @@ public function process(mixed $data, Operation $operation, array $uriVariables =
             $headers
         );
     }
+
+    private function getAllowedMethods(?string $resourceClass, ?string $currentUriTemplate): array
+    {
+        $allowedMethods = self::DEFAULT_ALLOWED_METHOD;
+        if (null !== $currentUriTemplate && null !== $resourceClass && $this->resourceClassResolver->isResourceClass($resourceClass)) {
+            $resourceMetadataCollection = $this->resourceCollectionMetadataFactory ? $this->resourceCollectionMetadataFactory->create($resourceClass) : new ResourceMetadataCollection($resourceClass);
+            foreach ($resourceMetadataCollection as $resource) {
+                foreach ($resource->getOperations() as $operation) {
+                    if ($operation->getUriTemplate() === $currentUriTemplate) {
+                        $allowedMethods[] = $operation->getMethod();
+                    }
+                }
+            }
+        }
+
+        return array_unique($allowedMethods);
```

Comment:
I don't think it's necessary because it's not possible to register two different routes for the same URL and the same method.

It would remove this as we are on the hot path.

To be honest, I would inline this code and remove this method. This will also allow you to set a variable (instead of running an O(n) operation) to check if POST is defined or not for Accept-Post, and this would remove a function call.

(Yes, these are mostly a micro optimization, but this also simplify the code IMHO).